Upgrade to symfony 6.4

hbagherzadegan opened the pull request.

hbagherzadegan committed:
upgrade to Symfony 6.3

- merge MacPaw/behat-messenger-context to our branch
- update composer package
- install rector
- run rector  -
 add rector to composer scripts
- adjust ci based on our services
- add friendsofphp/php-cs-fixer ,slevomat/coding-standard,squizlabs/php_codesniffer,roave/security-advisories

hbagherzadegan committed:
upgrade-to-symfony-6.4

- update composer
- update rector

JarJak commented:
we don't have to upgrade symfony in this repo, it is a fork so it would be best to keep our changes as minimum as possible to be able to sync with main repo. If you want to upgrade it to 6.4 please open PR to the main (vendor) repo.

hbagherzadegan committed:
- update rector to version 0.19.2
- remove deprecation from rector
- add CoversClass and UsesClass to tests
- update phpunit.xml.dist config

hbagherzadegan committed:
- fix symfony version on 6.4.* by adding extra to composer json
